### ISSUE TITLE
Default anonymous user doesn't pass `artifactory_user` validation

### ISSUE THREAD

**Issue by jcogilvie** [jfrog/terraform-provider-artifactory#265]
**Describe the bug**
My artifactory cloud installation generated an `anonymous` user by default.  If I try to adopt this user into terraform, it imports successfully but then fails validation on apply because it doesn't have an email.

**Requirements for and issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)

Import the user `anonymous` as below, then run an `apply` without adding/setting the email.  Note that it fails until you add one, even if there is no email set in the UI.

```hcl
resource "artifactory_user" "anonymous" {
  email             = "anonymous@foo.com" # this must be set, even though by default there is no value
  name              = "anonymous"
  profile_updatable = false
}
```
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
Cloud
- [x] Your version of terraform
1.0.11

**Expected behavior**
Apply should work as imported.  Either allow me to omit the email on `anonymous` or maybe create an `artifactory_anonymous_user` resource that doesn't require one.

**Comment by chb0github** [comment]
On the user creation page:
`ui/admin/management/users/new` email is a required field. So, RT created an invalid user - I mean, @alexhung it's possible it's accepted through curl/api but I don't think a blank email will fly. If not, the question is what to do about it? get the admin email somehow and use that?

**Comment by alexhung** [comment]
'anonymous' is a special, pre-defined user in Artifactory. The UI hides the username/email fields for that user. The API does not have any provision to manipulate an anonymous user either, since email is a required attribute in the payload schema.

Given these, I think the idea of a separate `artifactory_anonymous_user` resource to manage the anonymous user has merit. Can we make that resource 'read-only' somehow, i.e. only read returns data, and all other operations return 200/204? (Actually I *think* we can delete an `anonymous` user)

**Comment by chb0github** [comment]
You can create the resource and declare all fields as computed. I think that would do the trick

**Comment by jcogilvie** [comment]
I was able to update the email with the provider. It shows in the UI now.

I'm not sure that's consistent with it being read only as you've described.

**Comment by alexhung** [comment]
@jcogilvie So is this still an issue?

**Comment by jcogilvie** [comment]
Yes. The field is required even after importing a valid user (anon) which
doesn't have the field.

Setting it will 'take' iirc, but it's still an inconsistency in that if I
import the server's present configuration, I can't perform an apply.

(I think it ought to reject my attempt to add an email to anon, too,
probably.)

**Comment by chb0github** [comment]
@alexhung - so, to dove tail this with passwords, we would have 3 resources:
1. `anonymous_user`
2. `managed_user`
3. `unmanaged_user`

This isn't directly related to the problem of passwords but is related to users. 

My vote is to create a resource for this where all fields are computed

**Comment by alexhung** [comment]
@jcogilvie FYI, I'm working on this and I've added a new resource `artifactory_anonymous_user` where all attributes are optional. Importing the `anonymous` user from Artifactory involves the following HCL:

```hcl
resource "artifactory_anonymous_user" "anonymous" {
}
```

I've also fixed how we handle the TF state for `groups` which @maheshjfrog is also working on for #191. This means `terraform plan` after importing will show no state drift.

Of course if you alter the `artifactory_anonymous_user` attributes then all bets are off!
